### PR TITLE
DS-106 linting

### PR DESCRIPTION
**Pull request by AnthonyA44**
Main Reviewer: @Caterina-vanHerpt 

**Description & Background Context:**

Documentation for Linting and added a plugin for eslinting a11y 

**Where should the reviewer start?**

File name: readme.md, .eslintrc

**How should this be tested?**

Pull this branch and run commands specified under linting

**Are there any breaking changes?**

No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +53,10 @@
       "prettier --write"
     ]
```
Comment (Caterina-vanHerpt) [inline package.json:54]: if we add this block of code here under our lint staged then husky will automatically run eslint on staged files only so we don't have to rely on folks doing it manually each time before they commit:

`    "src/**/*.+(js|json|ts|tsx)": [
      "eslint"
    ]
`

Can you update the readme with this as well similar to how you described the prettier/husky/npm run lint?

File: package.json [hunk 2]
```diff
@@ -52,6 +53,10 @@
...
   },
+  "eslintIgnore": [
```
[the remark this answers is not in the document]
Comment (Caterina-vanHerpt) [reply package.json:56]: Also thinking we may want to do eslint on our test and story files, any reason we left these out?

**Review by Caterina-vanHerpt — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -52,6 +53,10 @@
...
+    "*.test.js",
+    "*.stories.js"
+  ],
   "eslintConfig": {
```
[the remark this answers is not in the document]
Comment (AnthonyA44) [reply package.json]: Moved ignore and config code into their respective files

File: package.json (hunk 1 above)
Comment (AnthonyA44) [reply package.json]: Updated package.json to include automated eslinting during commit process. Updated readme to also highlight this process

File: .eslintignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+*.test.js
+*.stories.js
```
Comment (Caterina-vanHerpt) [inline .eslintignore:2]: Just wanted to check in on what your thoughts were around excluding test and storie files from eslint?

**Review by Caterina-vanHerpt — APPROVED** [review]

**Pull request merged**